src/js/components/AppListComponent.jsx

```diff
@@ -98,16 +98,19 @@ var AppListComponent = React.createClass({
     var state = this.state;
     var appNodes = this.getAppNodes();
 
+    var pageIsLoading = state.fetchState === States.STATE_LOADING;
+    var noAppsOnPage = state.apps.length === 0;
```

Comment:
Nitpick o'day: I tend to prefer "positive" var names for booleans, as they make for easier reading. For examples, a few lines below we have to ask our poor little brains to decode the double negation: `!noAppsOnPage`.

What if we did: 

```
- var noAppsOnPage = state.apps.length === 0;
+ var pageHasApps = state.apps.length !== 0;

// And then
"hidden": pageIsLoading || pageHasApps
```

What do you think?